src/server/pfs/server/api_server.go

```diff
@@ -451,7 +451,8 @@ func (a *apiServer) putFileObj(ctx context.Context, objClient obj.Client, reques
 		})
 		return eg.Wait()
 	}
-	return put(request.File.Path, url.Path)
+	// Joining Host and Path to retrieve the full path after "scheme://"
+	return put(request.File.Path, path.Join(url.Host, url.Path))
```

Comment:
Yeah I tested it.  See the documentation here: https://golang.org/pkg/net/url/#URL